pyaedt/generic/ibis_reader.py

```diff
@@ -0,0 +1,748 @@
+        line : str
+            Current line content.
+
+        Returns
+        -------
+        str
+            Name of the component.
+
+        """
+        name = ""
+        name = line.replace("[Component]", "")
+        return name.strip()
+
+    # Pin
+    def make_pin_object(self, line, component_name, ibis):
+        """Extracts model's info.
+
+        Parameters
+        ----------
+        line : str
+            Current line content.
+        component_name : str
+            Name of the component.
+        ibis : :class:`pyaedt.generic.ibis_reader.Ibis`
+            ibis object containing all info.
+
+        Returns
+        -------
+        :class:`pyaedt.generic.ibis_reader.Pin`
+            Pin object.
+
+        """
+
+        current_string = ""
+
+        current_string = line.strip().replace("\t", " ")
+        pin_name = self.get_first_parameter(current_string)
+        pin = Pin(pin_name + "_" + component_name + "_" + ibis.name, ibis.circuit)
+        pin.short_name = pin_name
+        current_string = current_string[len(pin.name) + 1 :].strip()
+        pin.signal = self.get_first_parameter(current_string)
+
+        current_string = current_string[len(pin.signal) + 1 :].strip()
+        pin.model = self.get_first_parameter(current_string)
+
+        current_string = current_string[len(pin.model) + 1 :].strip()
+        pin.r_value = self.get_first_parameter(current_string)
+
+        current_string = current_string[len(pin.r_value) + 1 :].strip()
+        pin.l_value = self.get_first_parameter(current_string)
+
+        current_string = current_string[len(pin.l_value) + 1 :].strip()
+        pin.c_value = self.get_first_parameter(current_string)
+
+        return pin
+
+    def get_first_parameter(self, line):
+        """Gets first parameter string value.
+
+        Parameters
+        ----------
+        line : str
+            Current line content.
+
+        Returns
+        -------
+        str
+            First info extracted from the current line.
+
+        """
+
+        if line == "":
+            return ""
+
+        data = line.split(" ")
+        return data[0].strip()
+
+    def IsStartedWith(self, src, find, ignore_case=True):
+        """Verifies if a string content starts with a specific string or not.
+
+        Parameters
+        ----------
+        src : str
+            Current line content.
+        find : str
+            Current line content.
+        ignore_case : bool, optional
+            Case sensitive or not. The default value is ``True``.
+
+        Returns
+        -------
+        bool
+            ``True`` if the src string starts with the pattern.
+
+        """
+
+        if ignore_case == True:
+            if src[: len(find)].lower() == find.lower():
+                return True
+            else:
+                return False
+        else:
+            if src[: len(find)] == find:
+                return True
+            else:
+                return False
```

Comment:
This could be greatly simplified. Also, it appears this method only extends the built-in `startswith` by adding the ability to ignore case.

```suggestion
        if ignore_case:
            return src.lower().startswith(find.lower())
        return src.startswith(find)
```

Reply:
Great suggestion @akaszynski .It avoids to use a list.